storage/objects/main.go

```diff
@@ -96,6 +98,45 @@ func write(client *storage.Client, bucket, object string) error {
 	return nil
 }
 
+func list(client *storage.Client, bucket string) error {
+	ctx := context.Background()
+	// [START storage_list_files]
+	it := client.Bucket(bucket).Objects(ctx, nil)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		fmt.Println(attrs.Name)
+	}
+	// [END storage_list_files]
+	return nil
+}
+
+func listByPrefix(client *storage.Client, bucket, prefix string) error {
+	ctx := context.Background()
+	// [START storage_list_files_with_prefix]
+	// Prefix is a string that filters results to objects whose
+	// names begin with it. For example, "log-2016" will only list objects
+	// whose names will start with "log-2016".
```

Comment:
Add another comment that it's useful for emulating directory listings?

This page says
https://cloud.google.com/storage/docs/json_api/v1/objects/list

```
In conjunction with the prefix filter, the use of the delimiter parameter allows the list method to operate like a directory listing, despite the object namespace being flat. For example, if delimiter were set to "/", then listing objects from a bucket that contains the objects "a/b", "a/c", "d", "e", "e/f" would return objects "d" and "e", and prefixes "a/" and "e/".
```